App/Components/ReferralCode.tsx

```diff
@@ -59,7 +59,8 @@ const HIT_SLOP: Insets = {
 }
 
 interface Props {
-  referralCode: string
+  targetReferralCode: string
+  referralCode?: string
```

Comment:
The referral code input component now optionally takes a prop that is a referral code (set in the case the user is onboarding from opening the app from an invite link)